compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -1454,9 +1454,14 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
 impl<'ll> StaticBuilderMethods for Builder<'_, 'll, '_> {
     fn get_static(&mut self, def_id: DefId) -> &'ll Value {
         // Forward to the `get_static` method of `CodegenCx`
-        let s = self.cx().get_static(def_id);
+        let g = self.cx().get_static(def_id);
         // Cast to default address space if globals are in a different addrspace
-        self.cx().const_pointercast(s, self.type_ptr())
+        let g = self.cx().const_pointercast(g, self.type_ptr());
```

Comment:
llvm.threadlocal.address requires the argument to be a global value, it can't be a cast. So we should move this cast after the call to the intrinsic.

Reply:
Done!

Reply:
Well, that obviously didn't work. Quick question: Can globals emitted by rustc actually be in a different address space? And if so, how would I get its number for an `addrspacecast`?

Reply:
Ah right, I overlooked that that existed...